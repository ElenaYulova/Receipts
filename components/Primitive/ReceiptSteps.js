import React from 'react';
import PropTypes from 'prop-types';

import './ReceiptSteps.css';

// Компонент Этапы приготовления кулинарной книги. Переиспользуемый
export default class ReceiptSteps extends React.PureComponent {
    static propTypes = {
        name: PropTypes.string.isRequired,
        steps: PropTypes.string.isRequired,
        
    };

    
    render() {
        const name = this.props.name;
        let text = this.props.steps.split(".");
        text = text.map((part, i) => <p  key={i}>{part}</p> );
    return <div id = {name} className="receipt-steps">
        {text}        
    </div>
 }   
}