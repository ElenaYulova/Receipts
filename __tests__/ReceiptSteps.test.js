"use strict";

import React from 'react';
import renderer from 'react-test-renderer';
import ReceiptSteps from '../components/Primitive/ReceiptSteps';
import {shallow, configure} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';


configure({ adapter: new Adapter() });

describe('>>>ReceiptSteps --- Shallow Render REACT COMPONENTS',()=>{
    //shapshots
     
     
      const receiptComponent = <ReceiptSteps
      name = "name1"
      steps = "xdgfgjxhfjcgfh"
       />;
    
         
         const component = renderer.create(receiptComponent);
    
         it('renders receipt', () => {
           let componentTree = component.toJSON();
           expect(componentTree).toMatchSnapshot();
         });
    
         
    });