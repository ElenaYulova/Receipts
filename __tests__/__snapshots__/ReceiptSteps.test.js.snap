// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`>>>ReceiptSteps --- Shallow Render REACT COMPONENTS renders receipt 1`] = `
<div
  className="receipt-steps"
  id="name1"
>
  <p>
    xdgfgjxhfjcgfh
  </p>
</div>
`;
